lib/hanami/action/response.rb

```diff
@@ -81,6 +81,23 @@ def body=(str)
         end
       end
 
+      # Accept a symbolic status code
+      #
+      # @param code [Integer, Symbol] the status code
+      #
+      # @example
+      #   response.status = :unprocessable_entity
+      #
+      # @since 2.1.0
+      # @api public
+      def status=(code)
+        if code.is_a?(Symbol)
+          code = ::Rack::Utils::SYMBOL_TO_STATUS_CODE.fetch(code)
```

Comment:
I was relying on `KeyError` here but that's a fair point that it's an implementation detail. I was hesitant to create a new error type since there are so few but I'll do this